framework/PageDetails/PageDetailKeyValueList.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+  DescriptionList,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+
+type KeyValue = {
+  key: string;
+  value: string;
+};
+
+type KeyValueProps = {
+  keyValue: KeyValue[];
+  title: string;
+  keyColumn: string;
+  valueColumn: string;
+  renderValue?: (item: KeyValue) => JSX.Element;
+};
+
+export function PageDetailKeyValueList(props: KeyValueProps) {
+  const { keyValue, title, keyColumn, valueColumn, renderValue } = props;
+
+  const defaultRenderValue = (item: KeyValue) => <div data-cy={`item-value`}>{item.value}</div>;
+
+  const renderFunction = renderValue || defaultRenderValue;
+
+  return (
+    <>
+      <DescriptionListTerm data-cy={'key-value-list-title'}>{title}</DescriptionListTerm>
+      <DescriptionList isHorizontal style={{ paddingLeft: '20px' }}>
```

Comment:
padding left here is not optimal at small screens. might be better to have a section with heading and no left padding.